octoclient/xhrstreaming.py

```diff
@@ -0,0 +1,63 @@
+import random
+import string
+import requests
+import json
+from threading import Thread
+
+from urllib import parse as urlparse
+
+
+class XHRStreamingEventHandler:
+    @classmethod
+    def random_str(cls, length):
+        letters = string.ascii_lowercase + string.digits
+        return ''.join(random.choice(letters) for c in range(length))
+
+    def __init__(self, url, on_open=None, on_close=None, on_message=None):
+        self.on_open = on_open if callable(on_open) else lambda x: None
+        self.on_close = on_close if callable(on_close) else lambda x: None
+        self.on_message = on_message if callable(on_message) else lambda x, y: None
+
+        self.thread = None
+        self.socket = requests.Session()
+        parsed_url = urlparse.urlparse(url)
+        self.base_url = parsed_url.netloc
+        self.secure = True if parsed_url.scheme in ["wss", "https"] else False
+
+        server_id = str(random.randint(0, 1000))
+        session_id = self.random_str(8)
+        self.url = "{protocol}://" + "/".join((self.base_url, "sockjs", server_id, session_id, "xhr_streaming"))
+
+        self.url = self.url.format(protocol="https" if self.secure else "http")
+
+    def run(self):
+
+        self.thread = Thread(target=self.__xhr_streaming_run)
+        self.thread.daemon = True
+        self.thread.start()
+
+    def __xhr_streaming_run(self):
```

Comment:
Please, don't use double underscored in method names. There is no such thing as "private" in Python anyway, just use one underscore to indicate "non-API".